eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/io/VariantMongoAggregationReader.java

```diff
@@ -0,0 +1,176 @@
+
+    public static final String STUDY_ID_KEY = "SID";
+
+    public static final String CLUSTERED_VARIANT_VALIDATED_KEY = "RS_VALIDATED";
+
+    public static final String SUBMITTED_VARIANT_VALIDATED_KEY = "SS_VALIDATED";
+
+    public static final String ASSEMBLY_MATCH_KEY = "ASMM";
+
+    public static final String ALLELES_MATCH_KEY = "ALMM";
+
+    public static final String SUPPORTED_BY_EVIDENCE_KEY = "LOE";
+
+    public static final String MERGED_INTO_KEY = "CURR";
+
+    protected static final String DBSNP_SUBMITTED_VARIANT_ENTITY = "dbsnpSubmittedVariantEntity";
+
+    protected static final String ACCESSION_FIELD = "accession";
+
+    protected static final String REFERENCE_ASSEMBLY_FIELD = "asm";
+
+    protected static final String STUDY_FIELD = "study";
+
+    protected static final String CONTIG_FIELD = "contig";
+
+    protected static final String START_FIELD = "start";
+
+    protected static final String TYPE_FIELD = "type";
+
+    protected static final String REFERENCE_ALLELE_FIELD = "ref";
+
+    protected static final String ALTERNATE_ALLELE_FIELD = "alt";
+
+    protected static final String CLUSTERED_VARIANT_ACCESSION_FIELD = "rs";
+
+    protected static final String SS_INFO_FIELD = "ssInfo";
+
+    protected static final String VALIDATED_FIELD = "validated";
+
+    protected static final String ASSEMBLY_MATCH_FIELD = "asmMatch";
+
+    protected static final String ALLELES_MATCH_FIELD = "allelesMatch";
+
+    protected static final String SUPPORTED_BY_EVIDENCE_FIELD = "evidence";
+
+    private static final String RS_PREFIX = "rs";
+
+    protected String assemblyAccession;
+
+    protected MongoClient mongoClient;
+
+    protected String database;
+
+    protected MongoCursor<Document> cursor;
+
+    public VariantMongoAggregationReader(String assemblyAccession, MongoClient mongoClient, String database) {
+        this.assemblyAccession = assemblyAccession;
+        this.mongoClient = mongoClient;
+        this.database = database;
+    }
+
+    protected void aggregate(String collectionName) {
+        MongoDatabase db = mongoClient.getDatabase(database);
+        MongoCollection<Document> collection = db.getCollection(collectionName);
+        AggregateIterable<Document> clusteredVariants = collection.aggregate(buildAggregation())
+                                                                  .allowDiskUse(true)
+                                                                  .useCursor(true);
+        cursor = clusteredVariants.iterator();
+    }
+
+    abstract List<Bson> buildAggregation();
+
+    @Override
+    public List<Variant> read() throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
+        return cursor.hasNext() ? getVariants(cursor.next()) : null;
+    }
+
+    abstract List<Variant> getVariants(Document clusteredVariant);
+
+    protected VariantSourceEntry buildVariantSourceEntry(String study, Long mergedInto, String sequenceOntology,
```

Comment:
When 2 methods have exactly the same signature, the parameter that changes is usually added at the end.